### PR TITLE
Lua attribute player.ai_type

### PR DESCRIPTION
**Pull request by SimonHeimberg**
**Type of change**
New feature

**Before**
In a multiplayer game, it is not possible to to detect reliably if a player is controlled by an ai.
The property `wl.Game().interactive_player` only shows the human player on the current game instance.
Computer players are named "Computer N", but this is not a reliable source as it is translated in the hosts locale (#5991) and human players can choose such a name (partially prevented, see #5985).

**New behavior**
With the property `player.ai_type`, determining ai players is reliable. Making the exact type available to a script should not be a problem.

**Additional context**
Be aware the type of the ai player can change, probably only when a human player looses the connection and is replaced by an ai.
It is possible that a player is controlled by a human and an ai (by using `playerX.switchplayer()`). This can not be detected.

A use case is in [forum post ..48](https://www.widelands.org/forum/topic/5797/?page=1#post-40448)

### CODE REVIEW

File: src/scripting/lua_game.cc [hunk 1]
```diff
@@ -338,6 +339,19 @@ int LuaPlayer::set_hidden_from_general_statistics(lua_State* L) {
 	return 0;
 }
 
+/* RST
+   .. attribute:: ai_type
+
+      (RO) type of the ai controlling this player
+      ("normal", "weak", "very_weak", "empty", ..., "" if human controlled only)
+*/
```
Comment (Noordfrees) [inline src/scripting/lua_game.cc]:
```suggestion
/* RST
   .. attribute:: ai_type

      .. versionadded:: 1.2

      (RO) Type of the AI controlling this player
      ("normal", "weak", "very_weak", "empty"; "" if human controlled only)
*/
```

**Comment by SimonHeimberg** [comment]
Now the tests pass.
How can I verify that the netclient has this attribute correct?

**Comment by SimonHeimberg** [comment]
I could not yet practically test if the attribute is set in the game.

The setting `player.ai` is received and saved in netclient (see below, probably writes in player from game_settings.cc). So it hopefully is also available in game (on the netclient). But maybe `player.ai` (of src/logic/player.cc) stays empty.
https://github.com/widelands/widelands/blob/6ba442a57ac5a57d409488388e2d276c4eb3f5dc/src/network/gameclient.cc#L643

The type of ai is visible in the new game dialogue for the netclient. I guess it is still available in game.

**Comment by SimonHeimberg** [comment]
I finally managed to compile and have everything so I can start the executable.

The ai_type works as expected in multiplayer mode. Random ai players get a valid value in the game.
In single player mode random players seem to get the value "" (empty string), which is used for human players normally. As it is known that there is only one human player, it is still obvious, but unhandy.

**Comment by SimonHeimberg** [comment]
This could be included in 1.2 as is. It already adds helpful functionality. And it warns about what is not reliable.

The fixing of single player must be in a complete different place, probably in single player startup code. It could be investigated later.

**Comment by hessenfarmer** [comment]
@SimonHeimberg 
will try to test this in the next days

**Review by hessenfarmer — APPROVED** [review]
code LGTM. 
tested and working for:
- SP
- Local MP (host)
- Local MP (client)
 therefore lets have it

**Comment by hessenfarmer** [comment]
@bunnybot merge

**Pull request closed without merging**